packages/gatsby/src/redux/reducers/inference-metadata.ts

```diff
@@ -123,8 +82,8 @@ const incrementalReducer = (state = {}, action) => {
       // Can't simply add { fields: { [addedField]: node.fields[addedField] } }
       // because it will count `fields` key twice for the same node
       const previousFields = omit(node.fields, [addedField])
-      state[type] = deleteNode(state[type], { fields: previousFields })
-      state[type] = addNode(state[type], { fields: node.fields })
+      state[type] = deleteNode(state[type], { ...node, fields: previousFields })
+      state[type] = addNode(state[type], { ...node, fields: node.fields })
```

Comment:
These 2 changes have the potential to affect production code as we are now sending a full node instead of just fields. I based this on the fact that the action themselves (`deleteNode` and `addNode`) are already typed and coded to accept a full node as second parameter.

Reply:
This feels _correct_, so it's just a matter of if tests will pass correctly with this change. Thanks for figuring this out!

Reply:
Tests seem to pass nicely so it should be all good, I just wanted to make sure this change was seen.